pkg/handler/v2/external_cluster/external_cluster.go

```diff
@@ -256,7 +258,7 @@ func CreateEndpoint(
 				}
 			}
 
-			createdCluster, err := createOrImportAKSCluster(ctx, req.Body.Name, userInfoGetter, project, cloud, clusterProvider, privilegedClusterProvider)
+			createdCluster, err := createOrImportAKSCluster(ctx, req.Body.Name, userInfoGetter, project, spec.AKSClusterSpec, cloud, clusterProvider, privilegedClusterProvider)
```

Comment:
check for nil pointer value spec.AKSClusterSpec , same for eks, gke

Reply:
Alternatively , just pass the spec here as in createorImport method we are checking if its nil